src/pf.c

```diff
@@ -77,46 +80,84 @@ static void pf_submit(char const *type, char const *type_instance, uint64_t val,
 } /* void pf_submit */
 
 static int pf_read(void) {
+#ifdef __FreeBSD__
```

Comment:
This reads like two interleaved functions. It almost feels like one of those "Magic Eye" books ;)

Joking aside, this is a readability vs. maintainability trade-off. I think I would just duplicate the function, one for FreeBSD and one for other systems. This errs on the side of readability. Maintainability suffers because now the two functions could diverge, but I think they're simple enough that this is not a huge risk to take.

```c
#ifdef __FreeBSD__
static int pf_read(void) {
  // …
}
#else
static int pf_read(void) {
  // …
}
#endif
```

Reply:
Thanks for the feedback, it is exactly what I was looking for!  I will make that change.